dask/dataframe/tests/test_dataframe.py

```diff
@@ -3160,6 +3166,23 @@ def test_apply():
         ddf.apply(lambda xy: xy, axis="index")
 
 
+@pytest.mark.parametrize("convert_dtype", [None, True, False])
+def test_apply_convert_dtype(convert_dtype):
+    """Make sure that explicit convert_dtype raises a warning with pandas>=2.1"""
+    df = pd.DataFrame({"x": [1, 2, 3, 4], "y": [10, 20, 30, 40]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    kwargs = {} if convert_dtype is None else {"convert_dtype": convert_dtype}
+    if PANDAS_GT_210 and convert_dtype is not None:
+        ctx = pytest.warns(FutureWarning, match="the convert_dtype parameter")
+    else:
+        ctx = contextlib.nullcontext()
```

Comment:
(Not a block comment, feel free to ignore) this would be a good place to utilize this little `_check_warning` utility 

https://github.com/dask/dask/blob/b86786e3e1bd5be37a24d5b9743c6c043b9daffc/dask/utils_test.py#L154-L164